clientv3/auth.go

```diff
@@ -41,8 +42,11 @@ type Auth interface {
 	// UserChangePassword changes a password of a user.
 	UserChangePassword(ctx context.Context, name string, password string) (*AuthUserChangePasswordResponse, error)
 
-	// RoleAdd adds a new user to an etcd cluster.
+	// RoleAdd adds a new role to an etcd cluster.
 	RoleAdd(ctx context.Context, name string) (*AuthRoleAddResponse, error)
+
+	// RoleGrant grants a permission to a role.
+	RoleGrant(ctx context.Context, name string, readPerm bool, writePerm bool, prefix bool, key string) (*AuthRoleGrantResponse, error)
```

Comment:
I feel this interface is too complicated. 

How about

```
RoleGrant(ctx context.Context, name string, key string, type PermType)
```

And we define PermType as KeyR, KeyW, KeyRW, PrefixR, PrefixW, PrefixRW?